### ISSUE TITLE
BUG: accuracy of output structure

### ISSUE THREAD

**Issue by JaGeo** [materialsproject/atomate2#166]
**Describe the bug**
This is potentially related to a problem that occurs in the phonon workflow for one structure: BORN charges that do not reflect the correct symmetry. This then results in a wrong non-analytical term correction around Gamma.

I start with this POSCAR:
```
Mg3 Sb2
1.0
   2.3003714889113018   -3.9843602950772405    0.0000000000000000
   2.3003714889113018    3.9843602950772405    0.0000000000000000
   0.0000000000000000    0.0000000000000000    7.2813299999999996
Mg Sb
3 2
direct
   0.0000000000000000    0.0000000000000000    0.0000000000000000 Mg2+
   0.3333333333333333    0.6666666666666666    0.3683250000000000 Mg2+
   0.6666666666666667    0.3333333333333334    0.6316750000000000 Mg2+
   0.3333333333333333    0.6666666666666666    0.7747490000000000 Sb3-
   0.6666666666666667    0.3333333333333334    0.2252510000000000 Sb3-
````

Then, I run the following workfow:

```Python
from atomate2.vasp.jobs.core import DielectricMaker, TightRelaxMaker
from pymatgen.core.structure import Structure
from jobflow import run_locally
from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
from atomate2.vasp.powerups import update_user_incar_settings
from jobflow import Flow
flow=[]
struct=Structure.from_file("POSCAR")
job=TightRelaxMaker().make(structure=struct)
job2=DielectricMaker().make(structure=job.output.structure)
flow=Flow([job,job2],job2.output)
flow = update_user_incar_settings(flow, {"NPAR": 8}, class_filter=TightRelaxMaker)
run_locally(flow, create_folders=True)
```

The structure after the tight relaxation looks like this:
```
Mg3 Sb2                                 
   1.0000000000000000     
     2.2774694095435986   -3.9446927300134034    0.0000000000000000
     2.2774694095435986    3.9446927300134034   -0.0000000000000000
    -0.0000000000000000   -0.0000000000000000    7.1854093835379320
   Mg   Sb
     3     2
Direct
 -0.0000000000000000 -0.0000000000000000 -0.0000000000000000
  0.3333333333333357  0.6666666666666643  0.3675316159328170
  0.6666666666666643  0.3333333333333357  0.6324683840671832
  0.3333333333333357  0.6666666666666643  0.7761920492952012
  0.6666666666666643  0.3333333333333357  0.2238079507047988

  0.00000000E+00  0.00000000E+00  0.00000000E+00
  0.00000000E+00  0.00000000E+00  0.00000000E+00
  0.00000000E+00  0.00000000E+00  0.00000000E+00
  0.00000000E+00  0.00000000E+00  0.00000000E+00
  0.00000000E+00  0.00000000E+00  0.00000000E+00
```

The dielectric run then starts with:

```
Mg3 Sb2
1.0
   2.2774694100000001   -3.9446927299999999    0.0000000000000000
   2.2774694100000001    3.9446927299999999   -0.0000000000000000
  -0.0000000000000000   -0.0000000000000000    7.1854093800000003
Mg Sb
3 2
direct
  -0.0000000000000000   -0.0000000000000000   -0.0000000000000000 Mg
   0.3333333300000000    0.6666666700000000    0.3675316200000000 Mg
   0.6666666700000000    0.3333333300000000    0.6324683800000001 Mg
   0.3333333300000000    0.6666666700000000    0.7761920500000000 Sb
   0.6666666700000000    0.3333333300000000    0.2238079500000000 Sb
```

Clearly, this is less accurate. 

If someone already has a suspicion why this happens, I would be happy about any hint. Otherwise, I will go through the code and check how to fix this. I am currently not completely sure if this is leading to the symmetry problems in the BORN charges but it looks likely to me.

**Comment by JaGeo** [comment]
I currently think it is related to this line here: https://github.com/materialsproject/atomate2/blob/73b9b3e71be8addc237c9d6cfce3dd1c09aa2ca8/src/atomate2/vasp/schemas/calculation.py#L496
The final structure is taken from the vasprun.xml ... And, the vasprun.xml has fewer digits than the CONTCAR. @utf , should we instead read in the CONTCAR? I will in any cas try it locally.

**Comment by utf** [comment]
Hi @JaGeo, thanks for noticing this. It is a very important point! I hadn't realised the structure in the vasprun was written to such low precision. 

Let's read it from the CONTCAR as you suggest!

**Comment by JaGeo** [comment]
Yeah, I was actually also a bit surprised to see this low precision...

**Comment by utf** [comment]
Fixed in #167
